### PR TITLE
JMAPEmail: do not guess charset of valid UTF-8 with replacement chars

### PR DESCRIPTION
**Pull request by rsto**

### CODE REVIEW

File: Cassandane/Cyrus/JMAPEmail.pm [hunk 1]
```diff
@@ -22511,4 +22511,36 @@ EOF
...
+    :min_version_3_5 :needs_component_sieve :needs_component_jmap
+{
+    my ($self) = @_;
+    my $jmap = $self->{jmap};
+    my $imap = $self->{store}->get_client();
+
+    # MIME message contains a correctly encoded emoji and one UTF-8
+    # replacement character. The latter must not cause Cyrus to
+    # attempt guessing the source charset or report an encoding error.
+    open(my $F, 'data/mime/utf8-base64-replacement.eml') || die $!;
```
Comment (elliefm) [inline Cassandane/Cyrus/JMAPEmail.pm:22524]:
```suggestion
    open(my $F, '<', 'data/mime/utf8-base64-replacement.eml') || die $!;
```

trivial nit: it's a good habit to always use the three (or more) argument forms of `open()` -- though at the moment Cassandane is a bit of a hodgepodge

**Review by elliefm — APPROVED** [review]
Looks fine.  One trivial nit with suggestion inline

File: Cassandane/Cyrus/JMAPEmail.pm (hunk 1 above)
Comment (rsto) [reply Cassandane/Cyrus/JMAPEmail.pm]: Thanks, did not know that. Fixed.

**Comment by rsto** [comment]
Merged into https://github.com/cyrusimap/cyrus-imapd/pull/3753/commits/cd9838e8425e86561872e33cc392017652c17d6b

**Pull request closed without merging**